### PR TITLE
Ping Federate Data Connector

### PR DESCRIPTION
**Pull request by vu-socprime**
## Proposed Changes

  - add Ping Federate Data Connector

### CODE REVIEW

**Comment by preetikr** [comment]
@vu-socprime - Please have this data connector in Solutions folder (Solutions -> PingFederate -> Data Connectors - thanks.

**Comment by vu-socprime** [comment]
> @vu-socprime - Please have this data connector in Solutions folder (Solutions -> PingFederate -> Data Connectors - thanks.

@preetikr - Connector has been moved to Solutions/PingFederate/Data Connectors and parser to Solutions/PingFederate/Parsers

File: Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    "connectivityCriterias": [
+        {
+            "type": "IsConnectedQuery",
+            "value": [
+                "PingFederateEvent\n | summarize LastLogReceived = max(TimeGenerated)\n| project IsConnected = LastLogReceived > ago(30d)"
+            ]
+        }
+    ],
+    "availability": {
+        "status": 1
```
Comment (v-admahe) [inline Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json:35]: Status should be 2. Please update the status.

Thanks,
Aditi Maheshwari

Comment (v-admahe) [inline Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json:35]: IsPreview tag is missing.

Thanks,
Aditi Maheshwari

File: Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        {
+            "metricName": "Total data received",
+            "legend": "PingFederate Events",
+            "baseQuery": "PingFederateEvent"
+        }
+    ],
+    "sampleQueries": [
+        {
+            "description" : "Top 10 Rules", 
+            "query": "PingFederateEvent\n | summarize count() by RuleName\n | top 10 by count_"
```
Comment (v-admahe) [inline Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json:17]: Parser does not contain coulmn "RuleName". Please update the parser or query accordingly.

Thanks,
Aditi Maheshwari

**Review by v-admahe — CHANGES_REQUESTED** [review]

File: Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json [hunk 3]
```diff
@@ -0,0 +1,120 @@
+{
+    "id": "PingFederate",
+    "title": "PingFederate",
+    "publisher": "Ping Identity",
+    "descriptionMarkdown": "The [PingFederate](https://www.pingidentity.com/en/software/pingfederate.html) data connector provides the capability to ingest [PingFederate events](https://docs.pingidentity.com/bundle/pingfederate-102/page/lly1564002980532.html) into Azure Sentinel. Refer to [PingFederate documentation](https://docs.pingidentity.com/bundle/pingfederate-102/page/tle1564002955874.html) for more information.",
+    "additionalRequirementBanner": "This data connector depends on a parser based on Kusto Function to work as expected. Follow the steps to use this Kusto Function alias **PingFederateEvent** in queries and workbooks. [Follow steps to get this Kusto Function>](https://aka.ms/sentinel-PingFederate-parser)  ",
+    "graphQueries": [
+        {
+            "metricName": "Total data received",
+            "legend": "PingFederate Events",
```
[the remark this answers is not in the document]
Comment (vu-socprime) [reply Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json:10]: changed to PingFederateEvent

File: Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json (hunk 1 above)
Comment (vu-socprime) [reply Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json]: updated

Comment (vu-socprime) [reply Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json]: updated

File: Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json (hunk 2 above)
Comment (vu-socprime) [reply Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json]: fixed

File: Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json [hunk 4]
```diff
@@ -0,0 +1,121 @@
+{
+    "id": "PingFederate",
+    "title": "PingFederate",
+    "publisher": "Ping Identity",
+    "descriptionMarkdown": "The [PingFederate](https://www.pingidentity.com/en/software/pingfederate.html) data connector provides the capability to ingest [PingFederate events](https://docs.pingidentity.com/bundle/pingfederate-102/page/lly1564002980532.html) into Azure Sentinel. Refer to [PingFederate documentation](https://docs.pingidentity.com/bundle/pingfederate-102/page/tle1564002955874.html) for more information.",
+    "additionalRequirementBanner": "This data connector depends on a parser based on Kusto Function to work as expected. Follow the steps to use this Kusto Function alias **PingFederateEvent** in queries and workbooks. [Follow steps to get this Kusto Function>](https://aka.ms/sentinel-PingFederate-parser)  ",
+    "graphQueries": [
+        {
+            "metricName": "Total data received",
+            "legend": "PingFederateEvent",
```
Comment (v-admahe) [inline Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json:10]: Datatype name is PingFederate. So please update legend name to PingFederate.

Thanks,
Aditi Maheshwari

File: Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+                    "description": "Install the Microsoft Monitoring Agent on your Linux machine and configure the machine to listen on the necessary port and forward messages to your Azure Sentinel workspace. The CEF collector collects CEF messages on port 514 TCP.\n\n> 1. Make sure that you have Python on your machine using the following command: python -version.\n\n> 2. You must have elevated permissions (sudo) on your machine.",
+                    "instructions": [
+                        {
+                            "parameters": {
+                                "fillWith": [
+                                    "WorkspaceId",
+                                    "PrimaryKey"
+                                ],
+                                "label": "Run the following command to install and apply the CEF collector:",
+                                "value": "sudo wget https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/DataConnectors/CEF/cef_installer.py&&sudo python cef_installer.py {0} {1}"
```
Comment (v-admahe) [inline Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json:88]: -O cef_installer.py is missing in the command, please refer sample template https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Connector_CEF_template.json

File: Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+            "title": "3. Validate connection",
+            "description": "Follow the instructions to validate your connectivity:\n\nOpen Log Analytics to check if the logs are received using the CommonSecurityLog schema.\n\n>It may take about 20 minutes until the connection streams data to your workspace.\n\nIf the logs are not received, run the following connectivity validation script:\n\n> 1. Make sure that you have Python on your machine using the following command: python -version\n\n>2. You must have elevated permissions (sudo) on your machine",
+            "instructions": [
+                {
+                    "parameters": {
+                        "fillWith": [
+                            "WorkspaceId"
+                        ],
+                        "label": "Run the following command to validate your connectivity:",
+                        "value": "sudo wget https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/DataConnectors/CEF/cef_troubleshoot.py&&sudo python cef_troubleshoot.py  {0}"
```
Comment (v-admahe) [inline Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json:110]: Missing -O cef_troubleshoot.py is missing in the command, please refer sample template https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Connector_CEF_template.json

**Review by v-admahe — CHANGES_REQUESTED** [review]

**Comment by v-admahe** [comment]
@vu-socprime : Please move the logo  file in the solution folder. Please refer the link for folder structure https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/McAfeeePO/Data%20Connectors.

Thanks,
Aditi Maheshwari

**Comment by v-rucdu** [comment]
@vu-socprime : [Ubiquiti connector json](https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Solutions/Ubiquiti/Data%20Connectors/Connector_Ubiquiti_agent.json) is referring to a link (https://aka.ms/sentinel-UbiquitiUnifi-conf) resolving into ubiquiti.conf file, can you please provide it?

I am requesting here as the [Ubiquiti PR](https://github.com/Azure/Azure-Sentinel/pull/1918), is already merged. 

Thanks!!!

File: Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json (hunk 5 above)
Comment (vu-socprime) [reply Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json]: updated

File: Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json (hunk 6 above)
Comment (vu-socprime) [reply Solutions/PingFederate/Data Connectors/Connector_CEF_PingFederate.json]: updated

**Comment by vu-socprime** [comment]
> @vu-socprime : Please move the logo file in the solution folder. Please refer the link for folder structure https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/McAfeeePO/Data%20Connectors.
> 
> Thanks,
> Aditi Maheshwari

done

**Comment by v-rucdu** [comment]
@socprime : In Zscaler Private Access (ZAP) connector [PR](https://github.com/Azure/Azure-Sentinel/pull/2006), for one of the instruction steps you forgot to provide the version number can you please share it?

Below is the instruction step that is mentioned ZPA connector JSON,
**">**NOTE:** This data connector has been developed using Zscaler Private Access version: ____"**

Thanks!!!

**Comment by vu-socprime** [comment]
> @socprime : In Zscaler Private Access (ZAP) connector [PR](https://github.com/Azure/Azure-Sentinel/pull/2006), for one of the instruction steps version number is not provided can you please share it?
> 
> Below is the instruction step that is mentioned ZPA connector JSON,
> **">**NOTE:** This data connector has been developed using Zscaler Private Access version: ____"**
> 
> Asking for details here as that PR is already merged.
> 
> Thanks!!!

@v-rucdu - the version number is 21.67.1

**Comment by v-rucdu** [comment]
> > @socprime : In Zscaler Private Access (ZAP) connector [PR](https://github.com/Azure/Azure-Sentinel/pull/2006), for one of the instruction steps version number is not provided can you please share it?
> > Below is the instruction step that is mentioned ZPA connector JSON,
> > **">**NOTE:** This data connector has been developed using Zscaler Private Access version: ____"**
> > Asking for details here as that PR is already merged.
> > Thanks!!!
> 
> @v-rucdu - the version number is 21.67.1

Thanks @vu-socprime !!!!

**Review by v-jayakal — APPROVED** [review]

**Pull request merged**